call babelfish api on run parser and display errors

erizocosmico opened the pull request:
Calls the bblfsh server when the "run parser" button is clicked and passes the resultant AST to the UASTViewer. Also displays messages if something went wrong.

![screenshot_2017-06-28_17 37 54](https://user-images.githubusercontent.com/1312023/27646443-e89891d4-5c28-11e7-8b2a-9f3b3c94f0ef.png)

Serabe commented on src/App.js:
```diff
@@ -64,8 +67,38 @@ export default class App extends Component {
+                return err;
+              }
+
+              return null;
```
Extract this to an auxiliary function in the same file.

Serabe commented on src/App.js:
```diff
@@ -64,8 +67,38 @@ export default class App extends Component {
+
+          setTimeout(() => {
+            this.setState({ errors: [] });
+          }, 5000);
```
Remove this part. Is we are going to implement this, let's do it well. But we don't know what errors look like as of now.

erizocosmico replied:
Then they're there forever?

dpordomingo replied:
Since we have now a `Notification` component I'd turn it a kind of service, so you could trigger an error, and then it would handle its rise and dismiss timeout

Serabe commented on src/App.js:
```diff
@@ -83,21 +116,27 @@ export default class App extends Component {
+  }
+
+  get languageMode() {
+    return this.state.languages[this.currentLanguage].mode;
```
Great refactor.

Serabe commented on src/index.js:
```diff
@@ -5,6 +5,10 @@ import { injectGlobal } from 'styled-components';
+import { polyfill } from 'es6-promise';
+import 'isomorphic-fetch';
+
+polyfill();
```
```js
import { polyfill as es6PromisesPolyfill } from 'es6-promise';

es6PromisesPolyfill();
```

dpordomingo commented on src/App.js:
```diff
@@ -145,7 +177,23 @@ export default class App extends Component {
 }
+
+function normalizeError(err) {
+  if (typeof err === 'object' && err.hasOwnProperty('message')) {
```
Since we have a `service/api` to fetch the uast for a given code, I'd process the error there in order to return it in a expected way for the App instead of returning a raw server response, and handle here the error

dpordomingo commented:
aside: commits are not "signed-off-by"

dpordomingo commented:
I'd add a close button to the notification

erizocosmico commented:
Addressed all requested changes and added tests for API (since now contains more logic than just making an API call) and for notification) @dpordomingo @Serabe

Serabe commented on src/services/api.js:
```diff
@@ -0,0 +1,40 @@
+    return err;
+  }
+
+  return null;
```
What about just `err.toString()`?

erizocosmico replied:
`err` is something that came from the babelfish api and god knows what shape does it have. It's not an `Error`.

erizocosmico commented on src/App.js:
```diff
@@ -1,14 +1,14 @@
-import { Notifications, Error } from './components/Notifications';
-import languages from './languages';
-import * as api from './services/api';
+import React, { Component } from "react";
```
Yes, did a pass of prettier without the `single-quotes` flag lol

erizocosmico commented:
Fixed @dpordomingo

dpordomingo commented:
Did u pushed?

erizocosmico commented:
Yes, I amended the commit with a fix for the tests (forgot to accept the new snapshots)

dpordomingo commented:
Are we using double quotes for everything?

erizocosmico commented:
@dpordomingo nope, changed it
